docs/data/tree-view/tree-item-customization/LabelSlot.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { TreeItem2, TreeItem2Props } from '@mui/x-tree-view/TreeItem2';
+import { TreeViewBaseItem } from '@mui/x-tree-view/models';
+import { unstable_useTreeItem2 as useTreeItem2 } from '@mui/x-tree-view/useTreeItem2';
+
+type TreeItemWithLabel = {
+  id: string;
+  label: string;
+  secondaryLabel?: string;
+};
+
+export const MUI_X_PRODUCTS: TreeViewBaseItem<TreeItemWithLabel>[] = [
+  {
+    id: 'grid',
+    label: 'Data Grid',
+    children: [
+      {
+        id: 'grid-community',
+        label: '@mui/x-data-grid',
+        secondaryLabel: 'Community package',
+      },
+      {
+        id: 'grid-pro',
+        label: '@mui/x-data-grid-pro',
+        secondaryLabel: 'Pro package',
+      },
+      {
+        id: 'grid-premium',
+        label: '@mui/x-data-grid-premium',
+        secondaryLabel: 'Premium package',
+      },
+    ],
+  },
+  {
+    id: 'pickers',
+    label: 'Date and time pickers',
+
+    children: [
+      {
+        id: 'pickers-community',
+        label: '@mui/x-date-pickers',
+        secondaryLabel: 'Community package',
+      },
+      {
+        id: 'pickers-pro',
+        label: '@mui/x-date-pickers-pro',
+        secondaryLabel: 'Pro package',
+      },
+    ],
+  },
+  {
+    id: 'charts',
+    label: 'Charts',
+
+    children: [{ id: 'charts-community', label: '@mui/x-charts' }],
+  },
+  {
+    id: 'tree-view',
+    label: 'Tree View',
+    children: [{ id: 'tree-view-community', label: '@mui/x-tree-view' }],
+  },
+];
+
+interface CustomLabelProps {
+  children: string;
+  className: string;
+  secondaryLabel: string;
+}
+
+function CustomLabel({ children, className, secondaryLabel }: CustomLabelProps) {
+  return (
+    <div className={className}>
+      <Typography>{children}</Typography>
+      {secondaryLabel && (
+        <Typography variant="caption" color="secondary">
+          {secondaryLabel}
+        </Typography>
+      )}
+    </div>
+  );
+}
+
+const CustomTreeItem = React.forwardRef(function CustomTreeItem(
+  props: TreeItem2Props,
+  ref: React.Ref<HTMLLIElement>,
+) {
+  const { publicAPI } = useTreeItem2(props);
+
+  const item = publicAPI.getItem(props.itemId);
```

Comment:
This demo uses the `publicAPI`. I'm wondering if we should flag it, since we present this concept later on in the same page 🤔

Reply:
I don't think it's a problem
However I find it weird to use `useTreeItem2` just to retrieve `publicAPI`, it can cause quite a lot of computation, WDYT about exporting `publicAPI` from `useTreeItem2Utils`?

Reply:
yeah, I also find that weird 🙈 it's helpful when you use `useTreeItem2` for a more complex use case and you get the benefit of being able to retrieve the `publicAPI` too, but here it's indeed strange. So I think exporting it from `useTreeItem2Utils` makes sense 👌